### ISSUE TITLE
pylint and flake8 fighting over continued indentation

### ISSUE THREAD

**Issue by Kwpolska** [PyCQA/pylint#747]
With this input:

``` python
            to.append(t.format(name=pkg.name,
                               base=pkg.packagebase,
                               # […]
                               )
                      )
```

pylint complains:

```
************* Module pkgbuilder.utils
C:214, 0: Wrong continued indentation (remove 1 space).
                               )
                              |^ (bad-continuation)
C:215, 0: Wrong continued indentation (remove 1 space).
                      )
                     |^ (bad-continuation)
```

So, I “fixed” this.  flake8 now says:

```
pkgbuilder/utils.py:214:31: E124 closing bracket does not match visual indentation
pkgbuilder/utils.py:215:22: E124 closing bracket does not match visual indentation
```

Which linter is right?

**Comment by PCManticore** [comment]
In this case it seems that flake8 is right, with respect to what PEP 8 says:

"The closing brace/bracket/parenthesis on multi-line constructs may either line up under the first non-whitespace character of the last line of list or it may be lined up under the first character of the line that starts the multi-line construct".

**Comment by darkfeline** [comment]
Actually, both of your examples are wrong.

Wrong:

~~~ python
            to.append(t.format(name=pkg.name,
                               base=pkg.packagebase,
                               # […]
                               )
                      )
~~~

Wrong:

~~~ python
            to.append(t.format(name=pkg.name,
                               base=pkg.packagebase,
                               # […]
                              )
                     )
~~~

Right (lined up under the first non-whitespace character of the last line of list):

~~~ python
            to.append(t.format(name=pkg.name,
                               base=pkg.packagebase,
                               # […]
                               ))
~~~

Right (lined up under the first character of the line that starts the multi-line construct):

~~~ python
            to.append(t.format(name=pkg.name,
                               base=pkg.packagebase,
                               # […]
            ))
~~~

So pylint and flake8 are each catching different wrong indentations.

**Comment by cas--** [comment]
I am seeing the same issue with `flake8` passing and `pylint` flagging `bad-continuation` and incorrectly wanting to remove a space:

This is the current code that passes pep8:
```
            return c.connect(host, port, user, password
                             ).addCallback(on_connect, c, host_id
                                           ).addErrback(on_connect_failed, host_id)
```

Whereas this is what pylint wants to see which is incorrect:
```
            return c.connect(host, port, user, password
                            ).addCallback(on_connect, c, host_id
                                         ).addErrback(on_connect_failed, host_id)
```


```
pylint 1.5.5, 
astroid 1.4.5
Python 2.7.3 (default, Jun 22 2015, 19:43:34) 
[GCC 4.6.3]
```

**Comment by moylop260** [comment]
- Without remove a space:
 - pylint
<img width="814" alt="screen shot 2016-05-07 at 8 47 16 am" src="https://cloud.githubusercontent.com/assets/6644187/15092454/59c67bf6-1430-11e6-9045-54475c340129.png">
 - flake8
<img width="800" alt="screen shot 2016-05-07 at 8 52 35 am" src="https://cloud.githubusercontent.com/assets/6644187/15092482/1678008a-1431-11e6-8f73-d59debae8062.png">

- After remove a space:
 - pylint
<img width="777" alt="screen shot 2016-05-07 at 8 48 19 am" src="https://cloud.githubusercontent.com/assets/6644187/15092458/7d718e24-1430-11e6-83ad-e43f0bf75ffa.png">

 - flake8
<img width="821" alt="screen shot 2016-05-07 at 8 52 41 am" src="https://cloud.githubusercontent.com/assets/6644187/15092481/1677fa4a-1431-11e6-8d87-d5e43ff6954e.png">

Both style are valid for [pep-0008](https://www.python.org/dev/peps/pep-0008/)
```
 The closing brace/bracket/parenthesis on multi-line constructs may either line up under the first non-whitespace character of the last line of list, as in:

my_list = [
    1, 2, 3,
    4, 5, 6,
    ]
result = some_function_that_takes_arguments(
    'a', 'b', 'c',
    'd', 'e', 'f',
    )

or it may be lined up under the first character of the line that starts the multi-line construct, as in:

my_list = [
    1, 2, 3,
    4, 5, 6,
]
result = some_function_that_takes_arguments(
    'a', 'b', 'c',
    'd', 'e', 'f',
)
```

@PCManticore 
Should we allow both style or add a extra parameter to choose it?

**Comment by cas--** [comment]
To me it seem that pylint is incorrect here, further evidence is this exact same discussion of the issue in [yapf #26](https://github.com/google/yapf/issues/26)

Although it is not explicitly mentioned in PEP8 where the closing bracket should be from the example code in the docs. However this is my interpretation based on this code example:
```
# Aligned with opening delimiter.
foo = long_function_name(var_one, var_two,
                         var_three, var_four)
```
You can extrapolate it using the interpretation that the closing parenthesis is `under the first non-whitespace character of the last line of list` like so:
```
# Aligned with opening delimiter.
foo = long_function_name(var_one, var_two,
                         var_three, var_four
                         )
```
Therefore further extrapolating to this:
```
# Aligned with opening delimiter.
foo = long_function_name(var_one, var_two, var_three, var_four
                         )
```

**Comment by Pierre-Sassoulas** [comment]
Duplicate of #289, will be fixed by removing bad-continuation entirely in #3565.

**Comment by Pierre-Sassoulas** [comment]
`bad-continuation` has been removed in #3571, black or another formatter can help you with this better than Pylint
